### PR TITLE
Cleaned Subscription and fixed its isUnconfirmed method

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

File: src/Entities/Subscription.php [hunk 1]
```diff
@@ -84,198 +84,114 @@ class Subscription {
...
-	 * @param Address $address
-	 */
-	public function setAddress( $address ) {
-		$this->address = $address;
-	}
-
-	public function isUnconfirmed() {
-		return $this->getStatus() === self::STATUS_NEUTRAL;
+	public function isUnconfirmed(): bool {
+		return $this->getStatus() === self::STATUS_NEW;
```
Comment (gbirke) [inline src/Entities/Subscription.php]: Nope, moderated subscriptions (or subscriptions that have another status that will be added later) are also unconfirmed. More like

```php
return $this->getStatus() & self::STATUS_CONFIRMED === 0;
```

File: tests/unit/SubscriptionTest.php [hunk 2]
```diff
@@ -17,4 +17,16 @@ public function testGivenAHexConfirmationCode_itCanBeConvertedToBinary() {
...
+		$this->assertTrue( ( new Subscription() )->isUnconfirmed() );
+	}
+
+	public function testIsUnconfirmedReturnsFalseForConfirmedSubscriptions() {
+		$subscription = new Subscription();
+		$subscription->setStatus( Subscription::STATUS_CONFIRMED );
+
+		$this->assertFalse( $subscription->isUnconfirmed() );
+	}
+
```
Comment (gbirke) [inline tests/unit/SubscriptionTest.php]: `testIsUnconfirmedReturnsTrueForModeratedSubscriptions`?

File: src/Entities/Subscription.php (hunk 1 above)
Comment (JeroenDeDauw) [inline src/Entities/Subscription.php]: Well sure... I'm just retaining the previous behaviour here. Add new stuff would require new tests as well, which I'd rather as another PR

File: tests/unit/SubscriptionTest.php (hunk 2 above)
Comment (JeroenDeDauw) [inline tests/unit/SubscriptionTest.php]: You mean "subscriptionsPendingModeration"?

Anyway, same as for the other comment: out of scope here IMO

**Pull request merged**